spec/controllers/bookings_controller_spec.rb

```diff
@@ -27,15 +35,47 @@
   end
 
   before do
-    @user = FactoryBot.create(:user)
-    sign_in @user
+    sign_in user
   end
 
   describe "GET #index" do
     it "assigns all bookings to @bookings" do
-      booking = FactoryBot.create(:booking, user: @user)
       get :index
-      expect(assigns(:bookings)).to eq(@user.bookings)
+      expect(assigns(:bookings)).to eq(user.bookings)
+    end
+  end
+
+  # context 'with invalid attributes' do
+  #   it 'does not login for current user and redirect_to new_user_session_path' do
+  #     get :index
+  #     expect(response).to redirect_to(new_user_session_path)
+  #   end
+  # end
+
+  describe "GET #show" do
+    it "booking# @booking" do
+      get :show, params: { id: booking.id }
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  # describe "GET #new" do
+  #   it "assigns available seats excluding already booked ones" do
+  #     get :new, params: { bus_id: bus.id }
+  #     expected_seats = (1..bus.total_seats).to_a - [booking1.seat_no]
+  #     expect(assigns(:available_seats)).to eq(expected_seats)
+  #   end
+  # end
+
+  # it "renders the :new template" do
+  #   get :new, params: { bus_id: bus.id }
+  #   expect(response).to render_template(:new)
+  # end
```

Comment:
why you pushing commented code if you dont have any need?

Reply:
I need this code but i face issue that's why I commend this code